core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -408,6 +408,9 @@ private static void validateMatViewQuery(QueryModel model, String baseTableName)
                     throw SqlException.position(windowFuncColumn.getAst().position)
                             .put("window function on base table is not supported for materialized views: ").put(baseTableName);
                 }
+            } else {
+                throw SqlException.position(model.getModelPosition())
```

Comment:
I'm not sure if that's the right place to have this check. The base table may be referenced in a nested/joined/unioned model and we throw this exception without checking them.

Reply:
fixed in: https://github.com/questdb/questdb/pull/5619/commits/dd0b45a7f2c086c199d0ff04334cf6566ad3c449